langchain/utilities/loading.py

```diff
@@ -30,20 +68,22 @@ def try_load_from_hub(
     """Load configuration from hub.  Returns None if path is not a hub path."""
     if not isinstance(path, str) or not (match := HUB_PATH_RE.match(path)):
         return None
-    ref, remote_path_str = match.groups()
-    ref = ref[1:] if ref else DEFAULT_REF
+    source, ref, remote_path_str = match.groups()
     remote_path = Path(remote_path_str)
     if remote_path.parts[0] != valid_prefix:
         return None
     if remote_path.suffix[1:] not in valid_suffixes:
         raise ValueError("Unsupported file type.")
-
-    full_url = urljoin(URL_BASE.format(ref=ref), str(remote_path))
-    r = requests.get(full_url, timeout=5)
-    if r.status_code != 200:
-        raise ValueError(f"Could not find file at {full_url}")
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        file = Path(tmpdirname) / remote_path.name
-        with open(file, "wb") as f:
-            f.write(r.content)
-        return loader(str(file), **kwargs)
+    if source == "lc":
+        ref = ref[1:] if ref else DEFAULT_REF
+        full_url = urljoin(URL_BASE.format(ref=ref), str(remote_path))
+        r = requests.get(full_url, timeout=5)
+        if r.status_code != 200:
+            raise ValueError(f"Could not find file at {full_url}")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            file = Path(tmpdirname) / remote_path.name
+            with open(file, "wb") as f:
+                f.write(r.content)
+            return loader(str(file), **kwargs)
+    else:
+        return try_load_from_hf_hub(remote_path, loader, **kwargs)
```

Comment:
It might be safer here to check `elif source == "hf"` and explicitly raise a ValueError/return None if it's not the case.

Reply:
If it's not hf nor lc, then it won't match the regex and return `None` as per

```
if not isinstance(path, str) or not (match := HUB_PATH_RE.match(path)):
        return None
```

I could be explicit in the `elif` if preferred though :) but the error would not change things

Reply:
Oh ok, good :) 
(I still think explicit is better for readability but yeah, agree that the error will _never_ be trggerred)